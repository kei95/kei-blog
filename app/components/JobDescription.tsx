import RadioWaveButton from "~/assets/radioWaveCircleIcon";

export const JOB_DESCRIPTION_TEST_ID = "job_description";

export interface Job {
  title: string;
  period: string;
  description: string;
  roles: string[];
  tools: string[];
  isCurrentJob: boolean;
}

interface JobDescriptionProps {
  job: Job;
}

export default function JobDescription({ job }: JobDescriptionProps) {
  const { title, period, description, roles, tools, isCurrentJob } = job;

  return (
    <div className="max-w-screen-sm" data-testId={JOB_DESCRIPTION_TEST_ID}>
      <div className="flex flex-row">
        {/* Left side */}
        <div className="flex flex-col items-start">
          <RadioWaveButton
            animationState={isCurrentJob ? "ACTIVE" : "INACTIVE"}
            animationSize={20}
            buttonSize={8}
            className="-mt-1.5 -ml-6"
          />

          {/* vertical line */}
          <div className=" border-l-2 mt-1 ml-2.5 border-l-white h-full" />
        </div>

        {/* Right side */}
        <div className="flex flex-col">
          <div className="flex items-end">
            <h3 className="pr-4">{title}</h3>
            <p>{period}</p>
          </div>

          <p className="pb-3">{description}</p>

          {/* roles list */}
          <p className="font-bold text-textAccent">Key roles</p>
          <ul className="list-disc list-inside pb-3">
            {roles.map((role, idx) => (
              <li key={`${idx}_${role}`}>{role}</li>
            ))}
          </ul>

          {/* tools list */}
          <span className="font-bold text-textAccent">
            Tools:{" "}
            {tools.map((tool, idx) => (
              <span key={tool} className="pr-1 font-normal">
                {idx < tools.length - 1 ? `${tool},` : tool}
              </span>
            ))}
          </span>
        </div>
      </div>
    </div>
  );
}
