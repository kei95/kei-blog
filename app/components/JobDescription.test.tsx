import { render } from "@testing-library/react";
import JobDescription, { JOB_DESCRIPTION_TEST_ID } from "./JobDescription";
import type { Job } from "./JobDescription";

const mockedJob: Job = {
  title: "Generac Power Systems",
  period: "Apr 2021 - Now",
  description:
    "As a software developer, primary developing an app that allows a user to monitor his/her energy consumption and bill estimation.",
  roles: [
    "Implement the app's features with rich animations",
    "Design software architecture for new features from server side to client side",
    "Add new APIs to the server and maintain existing ones",
    "Led the team to add tests to the project and reduced the rate of bug report",
  ],
  tools: [
    "TypeScript",
    "React Naive",
    "React-native-testing-library",
    "Jest",
    "Java",
    "Maven",
    "Spring Boot",
    "AWS",
  ],
  isCurrentJob: true,
};

describe("<JobDescription / >", () => {
  test("given current job - it should render with given props and ANIMATED circle", () => {
    const { getByTestId } = render(<JobDescription job={mockedJob} />);

    expect(getByTestId(JOB_DESCRIPTION_TEST_ID)).toMatchSnapshot();
  });

  test("given previous job - it should render with given props and REGULAR circle", () => {
    const { getByTestId } = render(<JobDescription job={mockedJob} />);

    expect(getByTestId(JOB_DESCRIPTION_TEST_ID)).toMatchSnapshot();
  });
});
