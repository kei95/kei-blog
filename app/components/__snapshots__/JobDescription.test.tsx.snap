// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobDescription / > given current job - it should render with given props and ANIMATED circle 1`] = `
<div
  class="flex flex-row"
  data-testid="job_description"
>
  <div
    class="flex flex-col items-start"
  >
    <svg
      class="-mt-1.5 -ml-6"
      height="50"
      width="70"
    >
      <circle
        cx="50%"
        cy="50%"
        fill="none"
        r="0"
        stroke="#fff"
        stroke-width="2"
      >
        <animate
          attributeName="r"
          dur="3s"
          from="0"
          repeatCount="indefinite"
          to="20"
        />
        <animate
          attributeName="stroke-opacity"
          dur="3s"
          from="1"
          repeatCount="indefinite"
          to="0"
        />
      </circle>
      <circle
        cx="50%"
        cy="50%"
        fill="none"
        r="0"
        stroke="#fff"
        stroke-width="2"
      >
        <animate
          attributeName="r"
          begin="0.75s"
          dur="3s"
          from="0"
          repeatCount="indefinite"
          to="20"
        />
        <animate
          attributeName="stroke-opacity"
          begin="0.75s"
          dur="3s"
          from="1"
          repeatCount="indefinite"
          to="0"
        />
      </circle>
      <circle
        cx="50%"
        cy="50%"
        fill="none"
        r="0"
        stroke="#fff"
        stroke-width="2"
      >
        <animate
          attributeName="r"
          begin="1.5s"
          dur="3s"
          from="0"
          repeatCount="indefinite"
          to="20"
        />
        <animate
          attributeName="stroke-opacity"
          begin="1.5s"
          dur="3s"
          from="1"
          repeatCount="indefinite"
          to="0"
        />
      </circle>
      <circle
        class="fill-white stroke-white"
        cx="50%"
        cy="50%"
        r="8"
      />
    </svg>
    <div
      class=" border-l-2 mt-1 ml-2.5 border-l-white h-full"
    />
  </div>
  <div
    class="flex-wrap overflow-hidden"
  >
    <div
      class="flex flex-row "
    >
      <h3>
        Generac Power Systems
      </h3>
      <p
        class=" pl-4 text-textAccent"
      >
        Apr 2021 - Now
      </p>
    </div>
    <p
      class="pb-3"
    >
      As a software developer, primary developing an app that allows a user to monitor his/her energy consumption and bill estimation.
    </p>
    <p
      class="font-bold text-textAccent"
    >
      Key roles
    </p>
    <ul
      class="list-disc list-inside pb-3"
    >
      <li>
        Implement the app's features with rich animations
      </li>
      <li>
        Design software architecture for new features from server side to client side
      </li>
      <li>
        Add new APIs to the server and maintain existing ones
      </li>
      <li>
        Led the team to add tests to the project and reduced the rate of bug report
      </li>
    </ul>
    <span
      class="font-bold text-textAccent"
    >
      Tools: 
    </span>
    <span
      class="pr-1 font-normal"
    >
      TypeScript,
    </span>
    <span
      class="pr-1 font-normal"
    >
      React Naive,
    </span>
    <span
      class="pr-1 font-normal"
    >
      React-native-testing-library,
    </span>
    <span
      class="pr-1 font-normal"
    >
      Jest,
    </span>
    <span
      class="pr-1 font-normal"
    >
      Java,
    </span>
    <span
      class="pr-1 font-normal"
    >
      Maven,
    </span>
    <span
      class="pr-1 font-normal"
    >
      Spring Boot,
    </span>
    <span
      class="pr-1 font-normal"
    >
      AWS
    </span>
  </div>
</div>
`;

exports[`<JobDescription / > given previous job - it should render with given props and REGULAR circle 1`] = `
<div
  class="flex flex-row"
  data-testid="job_description"
>
  <div
    class="flex flex-col items-start"
  >
    <svg
      class="-mt-1.5 -ml-6"
      height="50"
      width="70"
    >
      <circle
        cx="50%"
        cy="50%"
        fill="none"
        r="0"
        stroke="#fff"
        stroke-width="2"
      >
        <animate
          attributeName="r"
          dur="3s"
          from="0"
          repeatCount="indefinite"
          to="20"
        />
        <animate
          attributeName="stroke-opacity"
          dur="3s"
          from="1"
          repeatCount="indefinite"
          to="0"
        />
      </circle>
      <circle
        cx="50%"
        cy="50%"
        fill="none"
        r="0"
        stroke="#fff"
        stroke-width="2"
      >
        <animate
          attributeName="r"
          begin="0.75s"
          dur="3s"
          from="0"
          repeatCount="indefinite"
          to="20"
        />
        <animate
          attributeName="stroke-opacity"
          begin="0.75s"
          dur="3s"
          from="1"
          repeatCount="indefinite"
          to="0"
        />
      </circle>
      <circle
        cx="50%"
        cy="50%"
        fill="none"
        r="0"
        stroke="#fff"
        stroke-width="2"
      >
        <animate
          attributeName="r"
          begin="1.5s"
          dur="3s"
          from="0"
          repeatCount="indefinite"
          to="20"
        />
        <animate
          attributeName="stroke-opacity"
          begin="1.5s"
          dur="3s"
          from="1"
          repeatCount="indefinite"
          to="0"
        />
      </circle>
      <circle
        class="fill-white stroke-white"
        cx="50%"
        cy="50%"
        r="8"
      />
    </svg>
    <div
      class=" border-l-2 mt-1 ml-2.5 border-l-white h-full"
    />
  </div>
  <div
    class="flex-wrap overflow-hidden"
  >
    <div
      class="flex flex-row "
    >
      <h3>
        Generac Power Systems
      </h3>
      <p
        class=" pl-4 text-textAccent"
      >
        Apr 2021 - Now
      </p>
    </div>
    <p
      class="pb-3"
    >
      As a software developer, primary developing an app that allows a user to monitor his/her energy consumption and bill estimation.
    </p>
    <p
      class="font-bold text-textAccent"
    >
      Key roles
    </p>
    <ul
      class="list-disc list-inside pb-3"
    >
      <li>
        Implement the app's features with rich animations
      </li>
      <li>
        Design software architecture for new features from server side to client side
      </li>
      <li>
        Add new APIs to the server and maintain existing ones
      </li>
      <li>
        Led the team to add tests to the project and reduced the rate of bug report
      </li>
    </ul>
    <span
      class="font-bold text-textAccent"
    >
      Tools: 
    </span>
    <span
      class="pr-1 font-normal"
    >
      TypeScript,
    </span>
    <span
      class="pr-1 font-normal"
    >
      React Naive,
    </span>
    <span
      class="pr-1 font-normal"
    >
      React-native-testing-library,
    </span>
    <span
      class="pr-1 font-normal"
    >
      Jest,
    </span>
    <span
      class="pr-1 font-normal"
    >
      Java,
    </span>
    <span
      class="pr-1 font-normal"
    >
      Maven,
    </span>
    <span
      class="pr-1 font-normal"
    >
      Spring Boot,
    </span>
    <span
      class="pr-1 font-normal"
    >
      AWS
    </span>
  </div>
</div>
`;
